sycl/include/sycl/ext/oneapi/properties/properties.hpp

```diff
@@ -305,7 +305,8 @@ struct all_props_are_keys_of<
     SyclT, ext::oneapi::experimental::properties<std::tuple<PropT, PropTs...>>>
     : std::bool_constant<
           ext::oneapi::experimental::is_property_key_of<PropT, SyclT>::value &&
-          all_props_are_keys_of<SyclT, PropTs...>()> {};
+          all_props_are_keys_of<SyclT, ext::oneapi::experimental::properties<
+                                           std::tuple<PropTs...>>>::value> {};
```

Comment:
I wonder if it would lead to better performance if we were to make `all_props_are_keys_of` a wrapper of some underlying check that takes a variable number of arguments, instead of continuously constructing smaller and smaller sub-`properties` that in turn need to instantiate a whole new version of `properties`.

Reply:
as it's compile-time I'm not really worried about performance. What I am worried about it readability. I'm not sure that an extra wrapping layer is going to improve that.

Reply:
> as it's compile-time I'm not really worried about performance. What I am worried about it readability. I'm not sure that an extra wrapping layer is going to improve that.

I disagree. Compile-time is a concern, which is partly why we switched to Boost for sorting and packing. As for readability, it will make it fill more lines, but I don't suspect it'll make it much less readable.

Reply:
done